### PR TITLE
 Apimon playbook for cbr

### PR DESCRIPTION
**Pull request by jmiskani**
created playbook for cbr

### CODE REVIEW

File: playbooks/scenario47_cbr.yaml [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+      - name: Create SG-Rule
+        openstack.cloud.security_group_rule:
+          security_group: "{{ test_security_group_name }}"
+          protocol: tcp
+          port_range_min: 22
+          port_range_max: 22
+          remote_ip_prefix: 0.0.0.0/0
+
+      - name: Get Availability zones
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:69]: you don't use it anywhere, so why you have it here?

File: playbooks/scenario47_cbr.yaml [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            - id: "{{ server_id }}"
+              type: "OS::Nova::Server"
+          action: "associate_resources"
+        register: vault
+
+      - name: Create restore point
+        opentelekomcloud.cloud.cbr_restore_point:
+          vault: "{{ test_vault_name }}"
+
+      - name: Query backups vor CBR vault
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:134]: for?

File: playbooks/scenario47_cbr.yaml [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        set_fact:
+          backup_id: "{{ backups.backups.0.id }}"
+
+      - name: Restore ECS server from backup
+        opentelekomcloud.cloud.cbr_backup:
+          name: "{{ backup_id }}"
+          server_id: "{{ server_id }}"
+          power_on: false
+
+      - name: Pause for 3 minutes to complete ECS restoration
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:149]: please create in otc ansible collection and in python-otcextensions issue requesting wait for restore completion method

**Review by vladimirhasko — CHANGES_REQUESTED** [review]

File: playbooks/scenario47_cbr.yaml (hunk 2 above)
Comment (jmiskani) [reply playbooks/scenario47_cbr.yaml:134]: corrected

File: playbooks/scenario47_cbr.yaml [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                state: absent
+
+            - name: Delete VPC
+              include_role:
+                name: vpc
+              vars:
+                state: absent
+                network_name: "{{ test_network_name }}"
+
+            - name: Delete KP
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:191]: Delete Keypair

File: playbooks/scenario47_cbr.yaml [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+          key_name: "{{ test_keypair_name }}"
+          network: "{{ test_network_name }}"
+        register: server
+
+      - name: Get server ID
+        set_fact:
+          server_id: "{{ server.server.id }}"
+
+      # Create CBR backup policy create / update
+      - name: Create Backup Policy
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:84]: Create CBR Backup Policy

File: playbooks/scenario47_cbr.yaml [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+          auto_expand: false
+          policy: "{{ test_policy_name }}"
+          billing:
+            consistent_level: "crash_consistent"
+            object_type: "server"
+            protect_type: "backup"
+            size: 40
+        register: vault
+
+      - name: Associate resouces CBR vault
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:117]: Associate resources to CBR vault

File: playbooks/scenario47_cbr.yaml [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+      - name: Query backups for CBR vault
+        opentelekomcloud.cloud.cbr_backup_info:
+          vault: "{{ test_vault_name }}"
+        register: backups
+        until: backups.backups[0].status == "available"
+        retries: 20 
+        delay: 30
+        ignore_errors: false
+
+      - name: Get backup ID
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:141]: Get CBR backup ID

File: playbooks/scenario47_cbr.yaml [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+      always:
+        # Clean created resourcesd
+        - block:
+
+            - name: Delete server
+              openstack.cloud.server:
+                state: absent
+                name: "{{ server_id }}"
+
+            - name: Delete Backup Vault
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:168]: Delete CBR server Backup Vault

File: playbooks/scenario47_cbr.yaml [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+      - name: Associate resouces CBR vault
+        opentelekomcloud.cloud.cbr_vault:
+          name: "{{ test_vault_name }}"
+          resources:
+            - id: "{{ server_id }}"
+              type: "OS::Nova::Server"
+          action: "associate_resources"
+        register: vault
+
+      - name: Create restore point
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:126]: Create CBR restore point

File: playbooks/scenario47_cbr.yaml [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+        until: backups.backups[0].status == "available"
+        retries: 20 
+        delay: 30
+        ignore_errors: false
+
+      - name: Get backup ID
+        set_fact:
+          backup_id: "{{ backups.backups.0.id }}"
+
+      - name: Restore ECS server from backup
```
Comment (vladimirhasko) [inline playbooks/scenario47_cbr.yaml:145]: Restore ECS server from CBR backup

**Review by vladimirhasko — CHANGES_REQUESTED** [review]

File: playbooks/scenario47_cbr.yaml (hunk 4 above)
Comment (jmiskani) [reply playbooks/scenario47_cbr.yaml:191]: updated

File: playbooks/scenario47_cbr.yaml (hunk 5 above)
Comment (jmiskani) [reply playbooks/scenario47_cbr.yaml:84]: updated

File: playbooks/scenario47_cbr.yaml (hunk 6 above)
Comment (jmiskani) [reply playbooks/scenario47_cbr.yaml:117]: updated

File: playbooks/scenario47_cbr.yaml (hunk 7 above)
Comment (jmiskani) [reply playbooks/scenario47_cbr.yaml:141]: updated

File: playbooks/scenario47_cbr.yaml (hunk 8 above)
Comment (jmiskani) [reply playbooks/scenario47_cbr.yaml:168]: updated

File: playbooks/scenario47_cbr.yaml (hunk 9 above)
Comment (jmiskani) [reply playbooks/scenario47_cbr.yaml:126]: updated

File: playbooks/scenario47_cbr.yaml (hunk 10 above)
Comment (jmiskani) [reply playbooks/scenario47_cbr.yaml:145]: updated

**Review by vladimirhasko — APPROVED** [review]

**Pull request merged**